matrix-json.c

```diff
@@ -128,3 +130,121 @@ const gchar *matrix_json_array_get_string_element(JsonArray *array,
 	element = matrix_json_array_get_element(array, index);
 	return matrix_json_node_get_string(element);
 }
+
+/* Arrays of name/node for an object */
+struct CanonicalObjectTmp {
+  guint total, count;
+  const char **names;
+  JsonNode **nodes;
+};
+
+/* Callback from matrix_canonical_json for each member of an object
+ * user_data is a pointer to a CanoicalObjectTmp
+ */
+static void canonical_object_child(JsonObject *object,
+        const gchar *member_name, JsonNode *member_node,
+        gpointer user_data)
+{
+  struct CanonicalObjectTmp *cot = user_data;
+  guint i;
+
+  g_assert_cmpuint(cot->count, <, cot->total);
+
+  /* The list needs to be with 'dictionary keys lexicographically sorted by unicode codepoint' */
+  for(i = 0; i < cot->count; i++) {
+    /* Is this the right comparison func? */
```

Comment:
OK, I've replaced this by a simple strcmp, I found a def that says:
   Sorting order: The chosen values of the leading bytes and the fact
   that the continuation bytes have the high-order bits first means
   that sorting a list of UTF-8 strings will produce the same order as
   sorting the equivalent UTF-32 strings
which I think means that a simple strcmp does the right thing.